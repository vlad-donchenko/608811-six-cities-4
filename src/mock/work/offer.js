const offer = [
  {
    id: 0,
    title: `Beautiful & luxurious apartment at great location`,
    picture: `apartment-01.jpg`,
    gallery: [`room.jpg`, `apartment-01.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `studio-01.jpg`, `apartment-01.jpg`],
    price: 120,
    type: `Apartment`,
    rating: 4,
    guests: 2,
    bedroomsCount: 2,
    advantages: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`, `Washing machine`, `Coffee machine`, `Dishwasher`, `Towels`, `Baby seat`, `Cabel TV`],
    isPremium: true,
    isBookmark: true,
    host: {
      name: `Angelina`,
      avatar: `avatar-angelina.jpg`,
      description: [`A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`, `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`],
      isPro: true
    },
    city: {
      name: `Amsterdam`,
      latitude: 52.38333,
      longitude: 4.9,
    },
    coordinates: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198,
      zoom: 12
    }
  },
  {
    id: 1,
    title: `Wood and stone place`,
    picture: `apartment-01.jpg`,
    gallery: [`room.jpg`, `apartment-01.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `studio-01.jpg`, `apartment-01.jpg`],
    price: 220,
    type: `Private room`,
    rating: 3,
    guests: 5,
    bedroomsCount: 3,
    advantages: [`Wi-Fi`, `Heating`, `Kitchen`, `Coffee machine`, `Dishwasher`, `Towels`, `Baby seat`, `Cabel TV`],
    isPremium: false,
    isBookmark: false,
    host: {
      name: `Angelina`,
      avatar: `avatar-angelina.jpg`,
      description: [`A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`, `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`],
      isPro: false
    },
    city: {
      name: `Amsterdam`,
      latitude: 52.38333,
      longitude: 4.9,
    },
    coordinates: {
      latitude: 52.369553943508,
      longitude: 4.85309666406198,
      zoom: 12
    }
  },
  {
    id: 2,
    title: `Canal View Prinsengracht`,
    picture: `apartment-01.jpg`,
    gallery: [`room.jpg`, `apartment-01.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `studio-01.jpg`, `apartment-01.jpg`],
    price: 150,
    type: `Apartment`,
    rating: 2,
    guests: 4,
    bedroomsCount: 2,
    advantages: [`Wi-Fi`, `Heating`, `Kitchen`, `Coffee machine`, `Dishwasher`, `Towels`],
    isPremium: false,
    isBookmark: false,
    host: {
      name: `Angelina`,
      avatar: `avatar-angelina.jpg`,
      description: [`A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`, `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`],
      isPro: true
    },
    city: {
      name: `Amsterdam`,
      latitude: 52.38333,
      longitude: 4.9,
    },
    coordinates: {
      latitude: 52.3909553943508,
      longitude: 4.929309666406198,
      zoom: 12
    }
  },
  {
    id: 3,
    title: `Nice, cozy, warm big bed apartment`,
    picture: `apartment-01.jpg`,
    gallery: [`room.jpg`, `apartment-01.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `studio-01.jpg`, `apartment-01.jpg`],
    price: 850,
    type: `Apartment`,
    rating: 5,
    guests: 5,
    bedroomsCount: 3,
    advantages: [`Wi-Fi`, `Coffee machine`, `Dishwasher`, `Towels`],
    isPremium: false,
    isBookmark: true,
    host: {
      name: `Angelina`,
      avatar: `avatar-angelina.jpg`,
      description: [`A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`, `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`],
      isPro: false
    },
    city: {
      name: `Amsterdam`,
      latitude: 52.38333,
      longitude: 4.9,
    },
    coordinates: {
      latitude: 52.3809553943508,
      longitude: 4.939309666406198,
      zoom: 12
    }
  }
];

export {offer};
