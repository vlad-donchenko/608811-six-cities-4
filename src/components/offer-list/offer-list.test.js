import React from "react";
import renderer from 'react-test-renderer';
import OfferList from "./offer-list.jsx";
import {offer} from "../../mock/test/offer";

it(`Should render OfferList`, () => {
  const tree = renderer.create(<OfferList offers={offer} onTitleClick={() => {}}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
