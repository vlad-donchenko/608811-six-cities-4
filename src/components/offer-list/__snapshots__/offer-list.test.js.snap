// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render OfferList 1`] = `
<div
  className="cities__places-list places__list tabs__content"
>
  <article
    className="cities__place-card place-card"
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            120
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button place-card__bookmark-button--active"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          luxurious apartment at great location - 1
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            520
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button place-card__bookmark-button--active"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          Wood and stone place - 10
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            150
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button "
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          View Prinsengracht - Wood
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            850
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button place-card__bookmark-button--active"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          Nice, cozy, warm big bed apartment - 10
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            750
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button "
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          Wood and stone place - 999
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
</div>
`;
